### PR TITLE
Modal updates and added shared App Project for SSB CSB

### PR DESCRIPTION
**Pull request by jbomhold3**
Moved App objects to SampleCore.
Added Ref support for modal. Actions Show(), Hide(),Toggle().
Added Escape Support For Modal.
I kept the IsOpen in the modal. But I think it honestly should be removed soon and file it as a breaking change. 

NOTE: PR in BlazorPrettyCode will need to be accepted and Nuget version updated before SSB will work right. See PR https://github.com/chanan/BlazorPrettyCode/pull/12

### CODE REVIEW

**Comment by chanan** [comment]
Looking at the BlazorPrettyCode PR now. I was going to merge it and not do a release yet since there are a bunch of other things I wanted to fix/add as well, but since you want to use it here, I will do a release after all.

**Comment by jbomhold3** [comment]
I can just uncomment the local ref and continue to debug that way. Debugging is far easier still on server side. Just wanted to make it clear this would have issues running the server side sample.

**Comment by chanan** [comment]
No, its ok, working on it now

**Comment by jbomhold3** [comment]
Left out this PR also adds escape support for dropdowns as well. I have it in toggle right now if it losses focus escape wont function.

File: src/BlazorStrap/Modal.razor [hunk 1]
```diff
@@ -48,7 +53,7 @@
         }
     }
     private bool _isOpen;
-    [Parameter]
```
Comment (chanan) [inline src/BlazorStrap/Modal.razor]: I see you added Hide/Show and removed IsOpen which makes sense, but I think we should still leave IsOpen as a Parameter and allow both styles. I can add it back in before I merge, just wanted to know if you see any problems with it.

Comment (jbomhold3) [reply src/BlazorStrap/Modal.razor]: Thought I added it back in was going to remove it then thought better of breaking peoples code

Comment (chanan) [reply src/BlazorStrap/Modal.razor]: No worries, adding it back, I also need to disable the Esc key when "keyboard: false" (or a better name) per #81 as mentioned here: https://getbootstrap.com/docs/4.3/components/modal/#options and then I will send out the new release. This PR is really great! Much better than the "alert" I had in SSB project :)

Comment (chanan) [reply src/BlazorStrap/Modal.razor]: Oops! You already added the IgnoreEscape! Sorry about that, didnt look before I wrote the above

Comment (jbomhold3) [reply src/BlazorStrap/Modal.razor]: I wanted to include the static files in the Core project but not really sure how to do it with client-side

Comment (chanan) [reply src/BlazorStrap/Modal.razor]: At some point I heard someone has a project that does that. I will try to find it

**Pull request merged**